azure_monitor/src/azure_monitor/trace.py

```diff
@@ -19,24 +20,24 @@ class AzureMonitorSpanExporter(BaseExporter, SpanExporter):
     def __init__(self, **options):
         super(AzureMonitorSpanExporter, self).__init__(**options)
 
-    def export(self, spans) -> SpanExportResult:
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
         envelopes = map(self.span_to_envelope, spans)
         envelopes_to_export = map(
             lambda x: x.to_dict(),
             tuple(self.apply_telemetry_processors(envelopes)),
         )
         try:
             result = self._transmit(envelopes_to_export)
-            if result == SpanExportResult.FAILED_RETRYABLE:
+            if result == utils.ExportResult.FAILED_RETRYABLE:
                 self.storage.put(envelopes, result)
-            if len(envelopes_to_export) < self.options.max_batch_size:
+            if len(list(envelopes_to_export)) < self.options.max_batch_size:
```

Comment:
How come we need make this a list?

Reply:
https://stackoverflow.com/questions/21572840/map-object-has-no-len-in-python-3

Reply:
I updated this logic because local storage was not being released correctly, updated logic to try to send all cached data when 200 response similar to node.js [SDK](https://github.com/microsoft/ApplicationInsights-node.js/blob/develop/Library/Sender.ts#L137)